packages/commons/src/rate-limiter/redisRateLimiter.ts

```diff
@@ -0,0 +1,108 @@
+import { Redis } from "ioredis";
+import { TenantId } from "pagopa-interop-models";
+import {
+  IRateLimiterRedisOptions,
+  RateLimiterRedis,
+  RateLimiterRes,
+} from "rate-limiter-flexible";
+
+import { match, P } from "ts-pattern";
+import { Logger } from "../logging/index.js";
+import { RateLimiter, RateLimiterStatus } from "./rateLimiterModel.js";
+
+export function initRedisRateLimiter(config: {
+  limiterGroup: string;
+  maxRequests: number;
+  rateInterval: number;
+  redisHost: string;
+  redisPort: number;
+  timeout: number;
+}): RateLimiter {
+  const redisClient = new Redis({
+    enableOfflineQueue: false,
+    host: config.redisHost,
+    port: config.redisPort,
+    commandTimeout: config.timeout,
+    // ^ If a command does not return a reply within a set number of milliseconds,
+    // a "Command timed out" error will be thrown.
+  });
+  // TODO Should we set reconnectOnError to reconnect on specific errors?
```

Comment:
yes if possible, thanks. This way we can reconnect for example on Redis restart or temporary connection lost